### PR TITLE
fix(desktop): stabilize long-session agent studio performance

### PR DESCRIPTION
**Pull request by Maxsky5**
## Summary
- isolate Agent Studio transcript rerenders by caching transcript rows off transcript-only data, memoizing the heavy row/card surfaces, and stabilizing the visible chat window reuse
- replace the experimental row-eviction transcript windowing with turn staging so long sessions stay responsive without scroll jumps, while disabling turn-level containment in the Tauri WKWebView shell where it caused unstable scrolling
- keep the git diff polling listener cleanup and restore git diff preloading after the branch audit so file diffs remain fast to open instead of paying the earlier branch-only regression

## Validation
- `bun run --filter @openducktor/desktop test`
- `bun run --filter @openducktor/desktop typecheck`
- `bun run --filter @openducktor/desktop lint`
- browser-mode validation against the live app with `agent-browser`
- manual Tauri retest confirming the transcript scroll jumps were resolved

## Notes
- The final branch diff intentionally keeps the transcript-focused fixes and the diff polling cleanup, but drops the earlier "remove diff preloading" direction because it slowed file diff display without helping the real root cause.
- Tauri uses macOS WKWebView rather than Chromium, so the branch now keeps turn containment in browser mode and disables it in the Tauri runtime only.

## Summary by CodeRabbit

* **New Features**
  * Turn-based transcript rendering and staged turn presentation for smoother, progressive display of long conversations.
* **Performance**
  * Improved memoization and identity-based caching for more stable rendering and faster navigation through large chat histories.
  * Windowing and scroll behavior refined to reduce unnecessary re-renders when transient session fields change.
* **Tests**
  * Added comprehensive unit and performance tests to ensure stability across runtimes and edge cases.

### CODE REVIEW

File: apps/desktop/src/components/features/agents/agent-chat/agent-chat-thread.tsx [hunk 1]
```diff
@@ -163,153 +472,87 @@ export function AgentChatThread({ model }: { model: AgentChatThreadModel }): Rea
...
+      if (cached) {
+        return cached;
+      }
+
+      const nextRef = registerRowElement(rowKey);
+      rowRefByKeyRef.current.set(rowKey, nextRef);
+      return nextRef;
+    },
+    [registerRowElement],
+  );
```
[the remark this answers is not in the document]
Comment (Maxsky5) [reply apps/desktop/src/components/features/agents/agent-chat/agent-chat-thread.tsx:487]: `registerRowElement` is already stable in the current code path, so the cached ref callbacks do not go stale. `useAgentChatRowMotion()` defines `registerRowElement` with `useCallback(..., [])`, and the thread-level regression coverage now locks the surrounding memoization/ref behavior. I kept the cache as-is to avoid extra churn in the hot transcript path.

**Comment by Maxsky5** [comment]
Follow-up on the remaining bot nits after the latest review-fix commit:

- Fixed: removed the dead `sessionSelectedModel` row prop, removed the redundant `requestAnimationFrame` unmount cleanup in `useAgentChatTurnStaging`, and isolated the `registerRowElement` test helper per test.
- Left as-is intentionally: the `AgentChatTurnGroup` memo-shape suggestion and `isTauriRuntime()` memo suggestion are both low-value optimizations without evidence of a remaining problem, and the optional explanatory-comment/docstring suggestions would add churn without improving correctness.

All project checks passed locally: `bun run test`, `bun run typecheck`, `bun run lint`, and `bun run build`.
